### ISSUE TITLE
Limit comparing files when it takes too much time

### ISSUE THREAD

**Issue by kodedninja** [beakerbrowser/beaker#1243]
When there are many changes in Preview mode, it would be nice to limit the time it takes to compare the published and unpublished versions. Maybe it's because of my weak computer, but every time I have to publish a lot of changes it takes about 10 minutes, because the whole view updates in every few minutes. So I hover the publish button, wait a few minutes, hover event is triggered, then click vividly until something happens. :sweat_smile:

**Comment by pfrazee** [comment]
Hah wow that's not good. How many files are in your dat? The preview mode, by default, only diffs the file tree. It *shouldn't* take that long.

**Comment by pfrazee** [comment]
Hmm! No `node_modules` right?

I wonder what we could do here. My first inclination is to improve the performance. We could try caching the diff results.

**Comment by pfrazee** [comment]
Basically that would require us to note the metadata (mtimes of dat version and FS version) and diff results for each file and then only check again if one of the timestamps changes. Should be pretty easy really.

**Comment by kodedninja** [comment]
Can my laptop be so bad? It happened two times since I've posted this. Approx. 30 minutes spent with waiting haha

**Comment by pfrazee** [comment]
That's definitely surprising. I'm going to try speeding up the file-tree diffs in the next release and we can see whether that helps.

**Comment by pfrazee** [comment]
Just pushed file-tree diffs to master. On my SSD, that gave me a 2x speedup. That said, when I tested very large file trees (a node_modules) I got really bad performance still, and that makes me think this is mostly related to https://github.com/beakerbrowser/beaker/issues/1250

I'm going to try to move the dat stack into its own process soon, to at least reduce browser stutter when dat has a slow-down.
